vpr/src/base/load_flat_place.cpp

```diff
@@ -0,0 +1,55 @@
+#include "globals.h"
+#include "load_flat_place.h"
+#include "clustered_netlist_utils.h"
+
+
+/* @brief Prints flat placement file entries for the atoms in one placed cluster. */
+static void print_flat_cluster(FILE* fp, ClusterBlockId iblk,
+                               std::vector<AtomBlockId>& atoms);
+
+static void print_flat_cluster(FILE* fp, ClusterBlockId iblk,
+                               std::vector<AtomBlockId>& atoms) {
+
+    auto& atom_ctx = g_vpr_ctx.atom();
+    t_pl_loc loc = g_vpr_ctx.placement().block_locs[iblk].loc;
+    size_t bnum = size_t(iblk);
+
+    for (auto atom : atoms) {
+        t_pb_graph_node* atom_pbgn = atom_ctx.lookup.atom_pb(atom)->pb_graph_node;
+        fprintf(fp, "%s  %d %d %d %d #%zu %s\n", atom_ctx.nlist.block_name(atom).c_str(),
+                                                loc.x, loc.y, loc.sub_tile,
+                                                atom_pbgn->flat_site_index,
+                                                bnum,
+                                                atom_pbgn->pb_type->name);
```

Comment:
I'd just do the size_t() case here so you don't create a new temporary variable bnum that is just an alias for iblk.  I imagine the size_t is just so you can print this number, so doing the size_t cast in the print is slightly easier to understand in my opinion.